src/inmanta/execute/runtime.py

```diff
@@ -649,6 +650,13 @@ def __init__(self, queue: QueueScheduler):
         self.queue.add_to_all(self)
         self.done = False
 
+    def _requeue_with_additional_requires(self, key: object, waitable: ResultVariable) -> None:
```

Comment:
Use [`typing.Hashable`](https://docs.python.org/3/library/typing.html#typing.Hashable) for `key`? It doesn't seem to reject values as strongly as you'd expect but at least it implies the intention to a user of the method.

Reply:
ah yes, but we would have to do that all over the place (this is a good place to start), thx